WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -81,121 +72,163 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+				NotifyMixableWalletLoaded(openedWallet);
 			}
-		}, stoppingToken);
+
+			// Notifies when a wallet no longer meets the criteria for participating in a coinjoin.
+			var closedWallets = trackedWallets.Where(x => !mixableWallets.ContainsKey(x.Key)).ToImmutableList();
+			foreach (var closedWallet in closedWallets.Select(x => x.Value))
+			{
+				//closedWallet.Cancel();
+				NotifyMixableWalletUnloaded(closedWallet);
+				trackedWallets.Remove(closedWallet.WalletName);
+			}
+			await Task.Delay(TimeSpan.FromSeconds(1), stoppingToken).ConfigureAwait(false);
+		}
+	}
 
 	private async Task MonitorAndHandleConjoinsAsync(CancellationToken stoppingToken)
 	{
 		// This is a shared resource and that's why it is concurrent. Alternatives are locking structures,
 		// using a single lock around its access or use a channel.
 		var trackedCoinJoins = new ConcurrentDictionary<string, CoinJoinTracker>();
 
-		var commandsHandlingTask = HandleCoinJoinCommandsAsync(trackedCoinJoins, stoppingToken);
-		var monitorCoinJoinTask = MonitorAndHandlingCoinJoinFinallizationAsync(trackedCoinJoins, stoppingToken);
+		var commandsHandlingTask = Task.Run(() => HandleCoinJoinCommandsAsync(trackedCoinJoins, stoppingToken).ConfigureAwait(false), stoppingToken);
+		var monitorCoinJoinTask = Task.Run(() => MonitorAndHandlingCoinJoinFinallizationAsync(trackedCoinJoins, stoppingToken).ConfigureAwait(false), stoppingToken);
 
 		await Task.WhenAll(commandsHandlingTask, monitorCoinJoinTask).ConfigureAwait(false);
 	}
 
-	private Task HandleCoinJoinCommandsAsync(ConcurrentDictionary<string, CoinJoinTracker> trackedCoinJoins, CancellationToken stoppingToken) =>
-		Task.Run(async () =>
+	private async Task HandleCoinJoinCommandsAsync(ConcurrentDictionary<string, CoinJoinTracker> trackedCoinJoins, CancellationToken stoppingToken)
+	{
+		var coinJoinTrackerFactory = new CoinJoinTrackerFactory(HttpClientFactory, RoundStatusUpdater, stoppingToken);
+
+		var trackedAutoStarts = new ConcurrentDictionary<Wallet, Task>();
+
+		while (!stoppingToken.IsCancellationRequested)
 		{
-			var coinJoinTrackerFactory = new CoinJoinTrackerFactory(HttpClientFactory, RoundStatusUpdater, stoppingToken);
+			var command = await CommandChannel.Reader.ReadAsync(stoppingToken).ConfigureAwait(false);
+			switch (command)
+			{
+				case StartCoinJoinCommand startCommand:
+					if (HandleStartCoinJoinCommand(startCommand, trackedCoinJoins, coinJoinTrackerFactory))
+					{
+						ScheduleRestartAutomatically(trackedAutoStarts, startCommand.Wallet, stoppingToken);
+					}
+					break;
+
+				case StopCoinJoinCommand stopCommand:
+					var walletToStop = stopCommand.Wallet;
+					if (trackedCoinJoins.TryGetValue(walletToStop.WalletName, out var coinJoinTrackerToStop))
+					{
+						coinJoinTrackerToStop.Cancel();
+					}
+					break;
+			}
+		}
+	}
 
-			while (!stoppingToken.IsCancellationRequested)
+	/// <returns>If restart should be scheduled or not.</returns>
+	private bool HandleStartCoinJoinCommand(
```

Comment:
This method does two different things:
* Asks whether a coinjoin should be rescheduled or not, and returns that.
* Starts a coinjoin.

It should be better to have tow different methods:
* `bool Should_Schedule_A_Restart()` and
* `void Create_And_Start_Coinjoin()`